src/js/jsx/search/SearchBar.jsx

```diff
@@ -68,8 +71,8 @@ define(function (require, exports, module) {
             }
 
             var idArray = id.split("_"),
-                type = idArray.length === 2 ? idArray[0] : "",
-                idInt = idArray.length === 2 ? parseInt(idArray[1]) : parseInt(id),
+                type = idArray.length > 0 ? idArray[0] : "",
+                idInt = idArray.length > 1 ? parseInt(idArray[1]) : parseInt(id),
```

Comment:
I gather that, in some cases, we're encoding an array of integers as a string? We should make whatever changes are necessary to `Datalist` to not do that. In particular, it would be fine if the `Datalist` change handler was applied to either a single number or an Array of numbers. It would be better for the handler to do this test than to bother with encoding and decoding an array of numbers as a string.

And, although this will be obviated by the above change, we prefer to use `parseNumber` from `js/util/math` instead of `window.parseInt` because the latter fails in weird ways. In particular, if the former fails it just returns null, whereas the latter returns `NaN`. (Fun JavaScript fact: `NaN !== NaN`)